src/process_utility.c

```diff
@@ -51,7 +52,31 @@ timescaledb_ProcessUtility(Node *parsetree,
 		return;
 	}
 
-	/* We don't support renaming hypertables yet so we need to block it */
+	/* Change the schema of hyper table */
+	if (IsA(parsetree, AlterObjectSchemaStmt))
+	{
+		AlterObjectSchemaStmt *alterstmt = (AlterObjectSchemaStmt *) parsetree;
+		Oid			relId = RangeVarGetRelid(alterstmt->relation, NoLock, true);
+
+		if (OidIsValid(relId))
+		{
+			Cache	   *hcache = hypertable_cache_pin();
+			Hypertable *hentry = hypertable_cache_get_entry(hcache, relId);
+
+			if (hentry != NULL && alterstmt->objectType == OBJECT_TABLE) {
+				executor_level_enter();
+				spi_hypertable_rename(hentry,
+									  alterstmt->newschema,
+									  NameStr(hentry->fd.table_name));
+				executor_level_exit();
+			}
+			cache_release(hcache);
+		}
+		prev_ProcessUtility((Node *) alterstmt, queryString, context, params, dest, completionTag);
```

Comment:
Agreed, but I was referring to the call to the standard processUtility, which is also handled by that function call (name slightly misleading). Just want to understand what part of the code handles the specific functionality.